tequilapi/endpoints/service.go

```diff
@@ -261,9 +246,53 @@ func AddRoutesForService(router *httprouter.Router, serviceManager ServiceManage
 	router.DELETE("/services/:id", serviceEndpoint.ServiceStop)
 }
 
-func toServiceRequest(req *http.Request) (sr serviceRequest, err error) {
-	err = json.NewDecoder(req.Body).Decode(&sr)
-	return sr, err
+func (se *ServiceEndpoint) toServiceRequest(req *http.Request) (serviceRequest, error) {
+	var jsonData struct {
+		ProviderID  string           `json:"providerId"`
+		ServiceType string           `json:"serviceType"`
+		Options     *json.RawMessage `json:"options"`
```

Comment:
Why `Options` should be a pointer here?

Reply:
Becauce it's optional argument - if You don't pass it, defaults is being used